packages/agents/src/AgentBase.ts

```diff
@@ -34,11 +37,34 @@ export interface AgentBaseConfig<TRunArgs> {
   timeout?: Timeout;
 }
 
-export abstract class AgentBase<TRunArgs, TAgentBaseContext extends AgentBaseContext> implements Agent<TRunArgs> {
+export class AgentBase<TRunArgs, TAgentBaseContext extends AgentBaseContext> implements Agent<TRunArgs> {
   constructor(
     public readonly config: AgentBaseConfig<TRunArgs>,
     protected context: TAgentBaseContext
-  ) {}
+  ) {
+    // Default functions that are added to every agent
+    const defaultFunctions = [
+      new ReadVariableFunction()
+    ];
+
+    // See which functions don't need to be added
+    const shouldAddDefault: Map<string, AgentFunctionBase<unknown> | undefined> = new Map(
```

Comment:
If you create a map of `this.config.functions` instead, you can do all this in 3 code blocks instead of 4

Reply:
Interesting, I'm not sure what this would look like though, care to elaborate?

Reply:
https://github.com/polywrap/evo.ninja/pull/335